### PR TITLE
Add sample for Unity mobile integration.

### PR DESCRIPTION
**Pull request by vincent-dfinity**
Add sample for Unity mobile integration.

### CODE REVIEW

**Comment by vincent-dfinity** [comment]
Hi @jessiemongeon1 , seems like I couldn't add you as a reviewer, so just ping you in the message. Please take a look when you're available. Thanks.

File: docs/samples/native-app-integration.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+- Use Deep Links or App Links to return to the game with the delegation
+- Use [ICP.NET](https://github.com/BoomDAO/ICP.NET) to call the functions provided by the canisters
+
+Below is a diagram that shows the workflow in detail.
+
+![Unity_ii_integration](./_attachments/unity-ii-integration.png)
+
+The steps in the diagram are described below:
+
+1. Generate an Ed25519 key pair from C# side
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:21]:
```suggestion
1. Generate an Ed25519 key pair from C# side.
```

File: docs/samples/native-app-integration.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+2. Open the dapp with the public key as a parameter
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:22]:
```suggestion
2. Open the dapp with the public key as a parameter.
```

File: docs/samples/native-app-integration.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+3. Authenticate with II
+4. Return the delegation
+5. Restore the game with the delegation via Deep/App Links
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:25]:
```suggestion
5. Restore the game with the delegation via Deep/App Links.
```

File: docs/samples/native-app-integration.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+6. Pass the delegation back to C#
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:26]:
```suggestion
6. Pass the delegation back to C#.
```

File: docs/samples/native-app-integration.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+7. Call the backend canister with DelegationIdentity provided by ICP.NET
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:27]:
```suggestion
7. Call the backend canister with DelegationIdentity provided by ICP.NET.
```

File: docs/samples/native-app-integration.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+8. Retrieve and show the principal id
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:28]:
```suggestion
8. Retrieve and show the principal ID.
```

File: docs/samples/native-app-integration.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+## Prerequisites
+
+- [x] Install the [IC SDK](../developer-docs/setup/install/index.mdx)
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:32]:
```suggestion
- [x] Install the [IC SDK](../developer-docs/setup/install/index.mdx).
```

File: docs/samples/native-app-integration.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+- [x] Clone [the examples repo](https://github.com/dfinity/examples/)
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:33]:
```suggestion
- [x] Clone [the examples repo](https://github.com/dfinity/examples/).
```

File: docs/samples/native-app-integration.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+- [x] Install [Unity](https://docs.unity3d.com/Manual/GettingStartedInstallingUnity.html)
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:34]:
```suggestion
- [x] Install [Unity](https://docs.unity3d.com/Manual/GettingStartedInstallingUnity.html).
```

File: docs/samples/native-app-integration.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+## Deployment
+
+### The Dapp
+
+The demo dapp mainly contains two parts:
+- Frontend
+  - II authentication integrated
```
Comment (jessiemongeon1) [inline docs/samples/native-app-integration.md:42]:
```suggestion
  - II authentication integrated.
```

**Review by jessiemongeon1 — APPROVED** [review]

**Pull request merged**